src/Background/Portal.vala

```diff
@@ -122,16 +122,14 @@ public class Background.Portal : Object {
         string[] commandline,
         AutostartFlags flags
     ) throws DBusError, IOError {
-        var filename = app_id;
-
         /* If the portal request is made by a non-flatpak application app_id will most of the time be empty
-         * We then use the commandline as a fallback for the autostart filename.
+         * We then use the commandline as a fallback.
          */
-        if (filename.strip () == "") {
-            filename = string.joinv ("-", commandline).replace ("--", "-");
+        if (app_id == "") {
+            app_id = commandline[0];
```

Comment:
i think we should validate that the first arg matches an app_id. since this can be things like `sudo` or `/usr/bin/python3`.

Reply:
Ah good point. How's this?

Reply:
LGTM